Sources/StreamChat/ChatClient.swift

```diff
@@ -227,10 +227,7 @@ public class ChatClient {
     
     /// The token of the current user. If the current user is anonymous, the token is `nil`.
     @Atomic var currentToken: Token?
-
-    /// In case of token expiration this property is used to obtain a new token
-    public var tokenProvider: TokenProvider?
-
```

Comment:
Could we just deprecate this, instead of totally removing it?

Reply:
@nuno-vieira when they are 2 different variables we have to manually keep them in sync: when any of it changes, another one needs to be changed as well. E.g. when a user connects as a `guest` but then assigns the new `tokenProvider`, what should we do with `userConnectionProvider`? ATM there's no way to understand what's the type of `userConnectionProvider` since it's just a closure.

It was a mistake to keep a `tokenProvider` separately from `userConnectionProvider` when they are essentially the same thing.

We could try to gracefully deprecate `tokenProvider` but it's gonna be a lot of boilerplate with the low-value outcome, I think. Customers leveraging `tokenProvider` will get a warning instead of a compilation error and by trying to fix it will end up with the implementation shown in PR description.

I'm against breaking changes in minor releases but this one seems like one customers will be happy about